### ISSUE TITLE
suggestions tooltip options

### ISSUE THREAD

**Issue by earmellin** [warpdotdev/Warp#5386]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

with the latest updates the tooltip with suggestions has filled up with entries related to shell functions that are not useful and make the tooltip less usable:

<img width="841" alt="image" src="https://github.com/user-attachments/assets/eeb95ab3-5db2-4407-9c5d-1bf6e5c447c5">

these tips disappear after hitting a space and the tooltip makes much more sense

<img width="841" alt="image" src="https://github.com/user-attachments/assets/600f7a85-41ef-424e-917b-e9b1792831f4">

### To reproduce

start to digit something

### Expected behavior

what you see in the second screenshot

### Operating system

MacOS

### Operating system and version

Sonoma 

### Shell Version

zsh 5.9 (x86_64-apple-darwin23.0)

### Current Warp version

v0.2024.10.08.08.02.stable_02

### Regression

Yes, this bug started recently or with an X Warp version

### Does this block you from using Warp daily?

No

### Is this an issue only in Warp?

Yes, I confirmed that this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hi @earmellin don't think much has changed for this feature in recent updates. You can try enabling the "Open completions as you type" in `Settings > Features > Editor`, this should help with pulling up the completions menu when you type in `yarn`+`SPACE`. I noticed you had no space on the screenshot with the shell functions. I hope this helps with your completion issue.

**Comment by earmellin** [comment]
hi @dannyneira this feature is already enabled, the problem is that until I complete the word "yarn" **and add a space** the suggestions are completely useless and misleading

**Comment by dannyneira** [comment]
@earmellin, I noticed that yarn isn't syntax highlighted on your second screenshot, indicating it may not be part of the PATH. If you run `which yarn` you should see the path for your yarn install. In my example, I got `/opt/homebrew/bin/yarn` so I added the homebrew bin directory to my PATH, and the yarn completion is loaded into new sessions (windows/tab/panes)

So in my example, I added the following to my `~/.zshrc` file:
```
# Homebrew
export PATH="/opt/homebrew/bin:$PATH"
```

![CleanShot 2024-10-17 at 10 33 15](https://github.com/user-attachments/assets/abcf48fa-6980-46e0-afbb-1569781f55fd)

Note how `yarn` is highlighted at the top of the completions menu. I hope this helps!

**Comment by earmellin** [comment]
I probably didn't explain myself well but this strange behavior of the tooltip is not specific to yarn but to any command, for example look this one related to ollama, only the first suggestion is significant the rest are completely useless:

<img width="411" alt="image" src="https://github.com/user-attachments/assets/82b2d171-26db-4e52-9066-cf4c9e1404eb">

or these related to brew:

<img width="411" alt="image" src="https://github.com/user-attachments/assets/ad9a395d-3356-4e1e-b4db-19dfe0cddc73">

**Comment by dannyneira** [comment]
Heard, this is due to the nature of the fuzzy search algorithm which is used across Warp. notice that those results contain the same letters, and in the case of brew, the same word. But the more relevant one should always be at the top. That being said, there is a related bug report claiming that the fuzzy search is showing too many results in CTRL-R [here](https://github.com/warpdotdev/Warp/issues/4174), so I take it this is an issue for other users as well.

**Comment by earmellin** [comment]
exactly! I noticed that it was related to the search algorithm, my request is really very similar to that of the other linked issue. In my case it would be enough to exclude all the entries categorized as "Shell function"
